KIPs/kip-149.md

```diff
@@ -244,6 +244,80 @@ func ReadActiveAddressFromRegistry(backend bind.ContractCaller, name string, num
 }
 ```
 
+#### JSON-RPC APIs
+
+The following JSON-RPC methods for the Klaytn node should be added to provide the records of registered system contracts.
+
+1. `klay_getActiveAddressFromRegistry`
+
+    - Parameters:
+        - `name`: the name of the system contract in string.
+        - `number`: (optional) integer or hexadecimal block number, or the string "pending" or "latest".
+    - Description: Returns the active address of the system contract registered as `name` if exists.
+    - Return: The address of the active system contract.
+    - Example
+        ```json
+        // Request
+        curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0", "method":"klay_getActiveAddressFromRegistry", "params":["KIP113", "latest"],"id":1}' http://localhost:8551
+        // Response
+        {
+            "jsonrpc":"2.0",
+            "id":1,
+            "result": "0x0000000000000000000000000000000000000402"
+        }
+
+        // Request - no active system contract
+        curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0", "method":"klay_getActiveAddressFromRegistry", "params":["KIP114", "latest"],"id":1}' http://localhost:8551
+        // Response
+        {
+            "jsonrpc":"2.0",
+            "id":1,
+            "error":{
+                "code":-32000,
+                "message":"no active address for KIP114"
+            }
+        }
+        ```
+2. `klay_getAllRecordsFromRegistry`
+
+    - Parameters:
+        - `name`: the name of the system contract in string.
+        - `number`: (optional) integer or hexadecimal block number, or the string "pending" or "latest".
+    - Description: Returns all records of the system contract registered as `name` if it has been registered.
+    - Returns:
+        - `Record[]`: An array of the records of the system contract.
+            - `Record`: A struct of the record with the following fields.
+                - `addr`: The address of the system contract.
+                - `activation`: The block number when the system contract will be activated.
```

Comment:
```suggestion
                - `activation`: The block number when the system contract is activated.
```

I'm not sure of this, but I think it would also include the addresses activated in the past.

Reply:
Thanks, applied in https://github.com/klaytn/kips/pull/159/commits/3364e36fd1e3ffdd671530229053a8928459f08f